modules/aggregation-layers/src/heatmap-layer/heatmap-layer.js

```diff
@@ -358,19 +348,16 @@ export default class HeatmapLayer extends CompositeLayer {
     const {
       triPositionBuffer,
       triTexCoordBuffer,
+      normalizedCommonBounds,
       visibleCommonBounds,
-      // scaledCommonBounds,
-      // commonBounds,
-      visibleWorldBounds,
-      worldBounds,
-      textureSize
+      visibleWorldBounds
     } = this.state;
 
-    const commonBounds = this._worldToCommonBounds(worldBounds, {
-      scaleToAspect: true,
-      width: textureSize * RESOLUTION,
-      height: textureSize * RESOLUTION
-    });
+    const {scale} = this.context.viewport;
+    const commonBounds = [
+      normalizedCommonBounds[0].map(x => x * scale),
+      normalizedCommonBounds[1].map(x => x * scale)
```

Comment:
@1chandu This is just unnecessary complexity. Can we make the bounds object flat?